exonum/src/blockchain/mod.rs

```diff
@@ -198,6 +238,7 @@ impl Blockchain {
             consensus: cfg.consensus,
             services: BTreeMap::new(),
             majority_count: None,
+            storage_version: storage::StorageVersion::current(),
```

Comment:
I'm not sure this will work as intended retrospectively. For example, imagine a situation that an auditor with the current storage version 2 first connects to the network that was initialized with the storage version 1. Then again, this *may* be the intended behavior; frankly speaking, I have *very* approximate understanding what should happen during version transition, and how multiple versions can be supported during the blockchain lifetime. Is this documented anywhere? @DarkEld3r

Reply:
This is definitely a concern for storage migration, but this out of scope for ECR-1423. To give you the sense of what are we trying to accomplish as of right now I want to quote the following lines from the task description:
> We need to store database version inside metadata table and panic if database contains unsupported version.

We agreed that **the migration** will be implemented *later*. For now we just have to implement a basic versioning mechanism. I can send you more information through a more private channel if you wish.

For now, the situation you described is, indeed, **intended behavior** (at least by me) - The blockchain runs only when all nodes have the same version.